### PR TITLE
PIM-3564: Add missing Models interfaces

### PR DESCRIPTION
**Pull request by jjanvier**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | N
| BC breaks?           | N
| CI currently passes? | Y
| Tests pass?          | Y
| Scenarios pass?      | Y
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | Y
| Fixed tickets        | -
| DB schema updated?   | -
| Migration script?    | -
| Doc PR               | -

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,6 @@ php:
 matrix:
     allow_failures:
         - php: "hhvm"
-        - php: "5.6"
```
Comment (nidup) [inline .travis.yml]: :+1:

File: spec/Pim/Bundle/CatalogBundle/Manager/AssociationTypeManagerSpec.php [hunk 2]
```diff
@@ -60,7 +60,7 @@ function it_throws_exception_when_save_anything_else_than_an_assoccation_type()
             ->shouldThrow(
                 new \InvalidArgumentException(
                     sprintf(
-                        'Expects an Pim\Bundle\CatalogBundle\Entity\AssociationType, "%s" provided',
+                        'Expects an "Pim\Bundle\CatalogBundle\Entity\AssociationTypeInterface", "%s" provided',
```
Comment (nidup) [inline spec/Pim/Bundle/CatalogBundle/Manager/AssociationTypeManagerSpec.php]: Pim\Bundle\CatalogBundle\Model\AssociationTypeInterface

File: spec/Pim/Bundle/CatalogBundle/Manager/AssociationTypeManagerSpec.php [hunk 3]
```diff
@@ -75,7 +75,7 @@ function it_throws_exception_when_remove_anything_else_than_an_assocation_type()
...
```
Comment (nidup) [inline spec/Pim/Bundle/CatalogBundle/Manager/AssociationTypeManagerSpec.php]: Model

File: spec/Pim/Bundle/CatalogBundle/Manager/AssociationTypeManagerSpec.php (hunk 2 above)
Comment (jjanvier) [inline spec/Pim/Bundle/CatalogBundle/Manager/AssociationTypeManagerSpec.php]: good catch !

File: src/Pim/Bundle/CatalogBundle/Model/GroupTypeInterface.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    /**
+     * Get the id
+     *
+     * @return int
+     */
+    public function getId();
+
+    /**
+     * Get code
+     * @return string $code
```
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Model/GroupTypeInterface.php]: missing blank line ;)

File: src/Pim/Bundle/CatalogBundle/Model/GroupTypeInterface.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    /**
+     * Get groups
+     *
+     * @return ArrayCollection
+     */
+    public function getGroups();
+
+    /**
+     * Get label
+     * @return string
```
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Model/GroupTypeInterface.php]: missing blank line

**Pull request merged**